tensorflow/contrib/tensorrt/convert/trt_optimization_pass.cc

```diff
@@ -251,8 +283,67 @@ tensorflow::Status TRTOptimizationPass::Optimize(
   cp.is_dyn_op = is_dynamic_op_;
   cp.cached_engine_batches = batches_;
   cp.max_cached_engines = max_cached_batches_;
+
+  if (save_input_graph_) {
```

Comment:
I believed we can do the debugging more easily in python level, e.g.
```
input_gdef = build_model()
with open('/tmp/input_gdef', 'w') as f:
  f.write(str(input_gdef))
output_gdef = trt.create_inference_graph(input_gdef)
with open('/tmp/output_gdef', 'w') as f:
  f.write(str(output_gdef))
```
So I think we don't need these logs and file outputs. Revert them?

Reply:
These are not the same. with trt.create_inference_graph, user has no control over which optimizers can be run. It is always fixed. Even if it is made configurable, above command will potentially return a slightly different graph than that will be executed. With these options, input and output graphs can be retrieved regardless of the position of TRTOptimizer in the optimization chain and closest to be executed.

Reply:
You're right that with current code it's not the same, but I still prefer not to add them, since once we add it it becomes a maintenance burden. I know sometimes it can make debugging easier but such functionality can be achieved much easier using python, with much less lines of code changes, for example, as you said by making the optimizer list (constfold, layout) configurable, and make it run only single round, etc. Or if you insist, at least we can make it generic by putting this logic to `MetaOptimizer::RunOptimizer()` so every optimizer can benefit from it.

Reply:
I would be quite happy to have this feature implemented in MetaOptimizer. Python path, unfortunately, is not always the same with session optimization pass.